lib/nodes/decorators/RepeatWhen.ts

```diff
@@ -26,7 +26,7 @@ export class RepeatWhen<S extends BlueshellState, E> extends Decorator<S, E> {
 		if (this.conditional(state, event, res)) {
 			// publish the tree before we reset so we can see the result
 			Action.treePublisher.publishResult(state, event, false);
-			clearChildEventSeen(this, state);
+			clearEventSeenRecursive(this, state);
```

Comment:
I think you wanted to call it with this.child, right?  Since this is being called after _beforeEvent of this, we don't want to reset it for this?

Reply:
This is the repeatWhen, which I left functionally unchanged. It is being called with this.child from When.

Reply:
Whoops, yes, all set here